gitlab/resource_gitlab_service.go

```diff
@@ -0,0 +1,405 @@
+package gitlab
+
+import (
+	"fmt"
+	"log"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	gitlab "github.com/xanzy/go-gitlab"
+)
+
+func resourceGitlabService() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceGitlabServiceCreate,
+		Read:   resourceGitlabServiceRead,
+		Update: resourceGitlabServiceUpdate,
+		Delete: resourceGitlabServiceDelete,
+
+		Schema: map[string]*schema.Schema{
+			"project": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"name": {
```

Comment:
Oh ok, now I get what you mean. What about just `service`? But you're right, `name` sounds too opened. Also, I'll add a check on this attribute to ensure the service exists.

Reply:
Ok, they internally use the `name` term:

```
  def self.available_services_names
    service_names = %w[
      asana
      assembla
[...]
```

cf. https://gitlab.com/gitlab-org/gitlab-ce/blob/master/app/models/service.rb#L247

The list of accepted values should be documented.
By the way, where's the doc? :p

Reply:
Or `slug`:

```
  describe 'POST /projects/:id/services/:slug/trigger' do
```

cf. https://gitlab.com/gitlab-org/gitlab-ce/blob/master/spec/requests/api/services_spec.rb#L105

Reply:
Haha good point, I just added the doc to the requirement.

So should we rather go for `name` or `service_name`? I feel like name is enough since it's the service resource. 

[ ] Add `services names` accepted values

Reply:
`slug` doesn't feel natural at all

Reply:
Having individual resources would eliminate this field.

Reply:
> I'm wondering if having individual resources, one for each service, to match the _unofficial_ go-gitlab client wouldn't be more natural and easier to document:
> 
> * `gitlab_service_slack`
> * `gitlab_service_jira`
> * etc.

Both would work. I honestly don't have a preference for one.